src/Storefront/Controller/DeprecatedContainerAccessTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Controller;
+
+use Psr\Container\ContainerInterface;
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ *
+ * @deprecated tag:v6.8.0 - Will be removed in 6.9.0
+ *
+ * @property ContainerInterface $container
+ *
+ * Bridge for legacy code accessing undeclared services.
+ */
+#[Package('framework')]
+trait DeprecatedContainerAccessTrait
+{
+    /**
+     * @deprecated tag:v6.8.0 - Declare services in getSubscribedServices()
+     */
+    protected function getServiceDeprecated(string $id): ?object
+    {
+        trigger_deprecation(
```

Comment:
we have our own deprecation handling, see https://github.com/shopware/shopware/blob/trunk/adr/2022-02-28-consistent-deprecation-notices-in-core.md#L8

please use that ;) actually this should already throw a phpstan error because you use @deprecated without the `Feature::triggerDeprecationOrThrow()` call